### PR TITLE
fix status and nav bar color

### LINKED ISSUE #19072: Fix Status and Nav Bar color
https://github.com/status-im/status-mobile/assets/17097240/2b924a66-e125-4c90-bf5b-57416802f0b1

### LINKED ISSUE #19366: switch theme event is called with wrong value while logging in
more details: https://discord.com/channels/1210237582470807632/1217173160416382986/1220703338438459462

### LINKED ISSUE #19367: Wrong status top bar color after switching device theme
Unfortunately, there are no exact steps. Sometimes it takes at least two or more attempts to switch the theme to reproduce this issue.

### Steps:
1. Switch device theme while being logged in the account
2. Pay attention at status top bar

### Actual result: 
Status top bar color is wrong. Status top bar color becomes correct only after app re-opening.

https://github.com/status-im/status-mobile/assets/52490791/da4314db-e86d-4cea-bb88-36e878bf7ed6

### Expected result: 
Status top bar color is correct when theme is switched

### Device:
Pixel 7a, Android 14

### logs:
[Status-debug-logs.zip](https://github.com/status-im/status-mobile/files/14722927/Status-debug-logs.zip)

### ENV:
Nightly [22 Mar 2024]

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-mobile/issues/19072
fixes https://github.com/status-im/status-mobile/issues/19367
fixes https://github.com/status-im/status-mobile/issues/19366

#### Android Navigation Bar
- For every light theme screen, the navigation bar should be light
- For every dark theme screen, the navigation bar should be dark
- For every light-themed bottom sheet, the navigation bar should be light
- For the home screen, the navigation bar should always be dark, due to the bottom tabs - check out bug video for reference https://github.com/status-im/status-mobile/issues/19072#issue-2163114567

#### For Status Bar
- For every light theme screen, the status bar should be dark
- For every dark theme screen, the status bar should be light
- When alert banners are visible, the status bar should be always light, no matter the screen
- Toasts should not break the status bar, as shown in https://github.com/status-im/status-mobile/issues/19072#issuecomment-1997484152

### Known Issues
- IOS - We are not able to change the iOS status bar theme color using the React native navigation library. So once theme is changed, app will still show old status color until relogin. In this PR we instead use the status bar api provided by react native itself. It works in the simulator, but not in iPhone 12. So it might or might not work on some/all devices. We can't fix this but please still let me know which is the case.
- Android - After login, the app theme should work as explained above. But as explained in https://github.com/status-im/status-mobile/pull/15596, the status bar theme that is provided at the time of logging in has high priority, and when we do operations like closing modal or going in the background, etc It reverts to that original status bar theme. Still, if you navigate to a different bottom tab or screen, it should fix itself. So PR partially fixes https://github.com/status-im/status-mobile/issues/19367

So basically, we can't fix the status bar color completely due to technical limitations. But PR should improve it and it should be better than develop. Please let me know if this is not the case.

status: ready

### CODE REVIEW

File: src/status_im/contexts/profile/login/events.cljs [hunk 1]
```diff
@@ -78,7 +78,7 @@
 
                   :else
                   [[:profile.settings/switch-theme-fx
-                    [(or (get-in db [:profile/profile :appearance])
```
Comment (Parveshdhull) [inline src/status_im/contexts/profile/login/events.cljs:81]: bug due to refactoring, as mentioned in https://discord.com/channels/1210237582470807632/1217173160416382986/1220703338438459462

File: src/status_im/navigation/effects.cljs [hunk 2]
```diff
@@ -1,29 +1,61 @@
...
+                  :shell-stack
+                  view-id)]
+    [rnn-status-bar rn-status-bar nav-bar-color comp-id]))
+
+(defn reload-status-nav-color-fx
+  [view-id]
+  (when (and (= @state/root-id :shell-stack) view-id)
+    (let [[rnn-status-bar rn-status-bar nav-bar-color comp-id] (get-status-nav-color view-id)]
+      (if platform/ios?
+        (rn/set-status-bar-style rn-status-bar true)
```
Comment (Parveshdhull) [inline src/status_im/navigation/effects.cljs:46]: this [doesn't work](https://github.com/status-im/status-mobile/issues/19072#issuecomment-1975156169) for android

**Review by alwx — APPROVED** [review]

**Comment by qoqobolo** [comment]
Hey @Parveshdhull , thanks a lot for the fixes! 
We'd like to include this PR in 2.28, so I've done some basic testing, and now the status bar and navigation bar work better indeed.

But there is one moment that differs from the release:

On iOS, the status bar is always dark (checked on iPhone 12 mini and SE). Watch the video, please:

https://github.com/status-im/status-mobile/assets/67952253/d24f7382-ed7f-4f32-9f9c-2fce5cbd7f71

On the release build it works fine - status bar color always changes to light. 
It also works as expected in PR for all other dark-themed screens. 

WDYT? Is this one of the consequences of using the status bar API provided by react native? Can it be fixed?

**Comment by qoqobolo** [comment]
As discussed, let's merge and cherry-pick it into 2.28 and fix/investigate the remaining issues separately as the behavior in basic flows looks really better now than in release 👍 Great job @Parveshdhull!
Would you mind creating a follow-up for the above iOS issue, please?
